pontoon/base/models.py

```diff
@@ -3306,6 +3306,12 @@ def save(self, update_stats=True, failed_checks=None, *args, **kwargs):
                     translation=t,
                 )
 
+            # Remove any TM entries of old translations that will get rejected.
+            # Must be executed before translations set changes.
+            TranslationMemoryEntry.objects.filter(
+                translation__in=approved_translations
+            ).delete()
```

Comment:
This is fine, but I did have to double-check that `approved_translations` is the right thing to filter by; while the name isn't strictly speaking misleading, it doesn't quite tell the whole story either.